jkube-kit/jkube-kit-spring-boot/src/test/java/org/eclipse/jkube/springboot/generator/AbstractSpringBootGeneratorIntegrationTest.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.springboot.generator;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.eclipse.jkube.generator.api.GeneratorContext;
+import org.eclipse.jkube.generator.api.GeneratorMode;
+import org.eclipse.jkube.kit.common.Assembly;
+import org.eclipse.jkube.kit.common.AssemblyFileSet;
+import org.eclipse.jkube.kit.common.Dependency;
+import org.eclipse.jkube.kit.common.JavaProject;
+import org.eclipse.jkube.kit.common.KitLogger;
+import org.eclipse.jkube.kit.common.Plugin;
+import org.eclipse.jkube.kit.config.image.ImageConfiguration;
+import org.eclipse.jkube.kit.config.image.build.BuildConfiguration;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class AbstractSpringBootGeneratorIntegrationTest {
+  abstract String getSpringBootVersion();
+  @TempDir
+  Path temporaryFolder;
+
+  private GeneratorContext context;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    context = GeneratorContext.builder()
+        .logger(new KitLogger.SilentLogger())
+        .project(createJavaProjectWithSpringBoot(getSpringBootVersion()))
+        .build();
+  }
+
+  @Test
+  void customize_whenGeneratorModeBuild_shouldGenerateOpinionatedImageConfiguration() throws JsonProcessingException {
+    // Given
+    SpringBootGenerator springBootGenerator = new SpringBootGenerator(context);
+    List<ImageConfiguration> images = new ArrayList<>();
+
+    // When
+    images = springBootGenerator.customize(images, false);
+
+    // Then
+    assertImageConfiguration(images, "8080");
+  }
+
+  @Test
+  void customize_whenGeneratorModeBuildAndApplicationPortOverridden_shouldGenerateOpinionatedImageConfiguration() throws JsonProcessingException {
```

Comment:
Try to not do generic tests with a single everything fits in assertion. This is not what I was trying to suggest.

The idea is that you have a specific test that for example states "SpringBoot Generator with xxx, generates an image with yyyy as from image" and so on.
I think I refactored both the Helidon and Quarkus generator to be explicit about what was being tested, please check those for reference.
The idea is that instead of having a single `assertImageConfiguration` you have a specific test for each behavior you want to enforce/verify (environment variables, ports, labels, artifacts, etc.).

Reply:
I have kept only specific assertions in other test cases. The big assertion is there only in default case.

Reply:
I've divided this test into smaller tests covering single aspects like image name, alias, port, env, assembly etc.